### PR TITLE
Add ability to view the progress for uploads and downloads

### PR DESCRIPTION
**Pull request by sujaygarlanka**
### Issue Link :link:
- https://jira.inside-box.net/browse/SDK-1181

### Goals :soccer:
- Add ability to view progress for uploads and downloads

### Implementation Details :construction:
- Add observer to progress object on a task

### Testing Details :mag:
- Manual Testing

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sujaygarlanka has signed the CLA. Thanks for the pull request!

File: Sources/Network/BoxNetworkAgent.swift [hunk 1]
```diff
@@ -143,7 +143,24 @@ public class BoxNetworkAgent: NSObject, NetworkAgentProtocol {
...
+        // Key value observer: Observer attaches to Progress object on task. Every time the Progress object updates,
+        // progressHandler is called with the Progress object passed in. progressHandler calls the request's progress closure and passes in the updated
+        // progress object.
+        var observation: NSKeyValueObservation?
+        let progressHandler: (Progress, NSKeyValueObservedChange<Double>) -> Void = {
+            progress, _ in
+            request.progress(progress)
+            if progress.fractionCompleted >= 1 {
+                observation?.invalidate()
+                observation = nil
```
Comment (jlawton) [inline Sources/Network/BoxNetworkAgent.swift]: I wouldn't block on this, but I think it'll be worth testing the case that a task is cancelled before completion, to ensure that the observation is cleaned up.

Comment (sujaygarlanka) [reply Sources/Network/BoxNetworkAgent.swift]: Made a change where I invalidate the observation in callback of the URLSessionTask. So in the case where the task is cancelled, the callback will be called with an error and the observation will be invalidated.

File: SwiftSDK.playground/contents.xcplayground [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (PJSimon) [inline SwiftSDK.playground/contents.xcplayground]: Remove playground file via rollback

**Comment by PJSimon** [comment]
Closing PR for now, and we will address issues in comments later.

**Pull request closed without merging**